File: catalogue-service/src/main/java/org/nc/entity/Product.java
package org.nc.entity;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.DBRef;
import org.springframework.data.mongodb.core.mapping.Document;

@Document
@Data @AllArgsConstructor @NoArgsConstructor @ToString(exclude = "category")
public class Product {
    @Id
    private String id;
    private String name;
    private double price;
    private boolean promotion;
    private boolean available;
    private String description;
    private String image;
    private double stock;

    @DBRef
    private Category category;
}

File: angular-front-end/src/app/model/product.ts
import { Category } from './category';

export class Product {
  id: string | null = null;
  name: string = '';
  price: number = 0;
  promotion: boolean = false;
  available: boolean = true;
  description: string = '';
  image: string = '';
  stock: number = 0;
  category: Category = new Category();
  constructor() {
  }
}
